[BUG] NullPointerException when restoring searchable snapshot created in Opensearch without shallow snapshots support

### Describe the bug

After upgrading from OpenSearch 2.8 to OpenSearch 2.19, I can no longer restore searchable snapshots that were created using earlier Opensearch version. Restore process fails with a NullPointerException related to the isRemoteStoreIndexShallowCopyEnabled() field.

It seems that snapshots created with old OpenSearch versions (before 2.10???) do not contain the `remote_store_index_shallow_copy` field, and the current code does not handle its absence gracefully. This leads to a failure when restoring with storage_type: "remote_snapshot".

I can successfully restore snapshot
```
{
  "snapshots" : [
    {
      "snapshot" : "logstash-2025.03.01",
      "uuid" : "WUSPPQiuQc2od8PMyR5KTQ",
      "version_id" : 136407827,
      "version" : "2.19.0",
      "remote_store_index_shallow_copy" : false,
      "indices" : [
        "logstash-2025.03.01"
      ],
      "data_streams" : [ ],
      "include_global_state" : false,
      "state" : "SUCCESS",
      "start_time" : "2025-03-03T04:08:32.687Z",
      "start_time_in_millis" : 1740974912687,
      "end_time" : "2025-03-03T04:09:57.917Z",
      "end_time_in_millis" : 1740974997917,
      "duration_in_millis" : 85230,
      "failures" : [ ],
      "shards" : {
        "total" : 5,
        "failed" : 0,
        "successful" : 5
      }
    }
  ]
}
```
But cannot restore
```
{
  "snapshots" : [
    {
      "snapshot" : "logstash-2025.02.01",
      "uuid" : "CX4nReZiTrqtxZQsoh9wYg",
      "version_id" : 136297827,
      "version" : "2.8.0",
      "indices" : [
        "logstash-2025.02.01"
      ],
      "data_streams" : [ ],
      "include_global_state" : false,
      "state" : "SUCCESS",
      "start_time" : "2025-02-03T04:10:10.401Z",
      "start_time_in_millis" : 1738555810401,
      "end_time" : "2025-02-03T04:11:34.642Z",
      "end_time_in_millis" : 1738555894642,
      "duration_in_millis" : 84241,
      "failures" : [ ],
      "shards" : {
        "total" : 5,
        "failed" : 0,
        "successful" : 5
      }
    }
  ]
}
```
### Related component

_No response_

### To Reproduce

1.    Create a searchable snapshot using OpenSearch 2.8.

2.    Upgrade to OpenSearch 2.19.

3.    Attempt to restore the snapshot using the following request:
```
POST /_snapshot/s3_arch_repository/logstash-2025.02.01/_restore" -H 'Content-Type: application/json' -d'
    {
      "ignore_unavailable": true,
      "include_global_state": false,
      "include_aliases": false,
      "partial": false,
      "storage_type": "remote_snapshot"
    }'
```
4.    Observe the NullPointerException
```
{
  "error": {
    "root_cause": [
      {
        "type": "null_pointer_exception",
        "reason": "Cannot invoke \"java.lang.Boolean.booleanValue()\" because the return value of \"org.opensearch.snapshots.SnapshotInfo.isRemoteStoreIndexShallowCopyEnabled()\" is null"
      }
    ],
    "type": "null_pointer_exception",
    "reason": "Cannot invoke \"java.lang.Boolean.booleanValue()\" because the return value of \"org.opensearch.snapshots.SnapshotInfo.isRemoteStoreIndexShallowCopyEnabled()\" is null"
  },
  "status": 500
}
```
### Expected behavior

Restoring searchable snapshots created in 2.8.0 is possible in Opensearch 2.19.0.

### Additional Details

**Plugins**
opensearch-alerting
opensearch-anomaly-detection
opensearch-asynchronous-search
opensearch-cross-cluster-replication
opensearch-custom-codecs
opensearch-flow-framework
opensearch-geospatial
opensearch-index-management
opensearch-job-scheduler
opensearch-knn
opensearch-ltr
opensearch-ml
opensearch-neural-search
opensearch-notifications
opensearch-notifications-core
opensearch-observability
opensearch-performance-analyzer
opensearch-reports-scheduler
opensearch-security
opensearch-security-analytics
opensearch-skills
opensearch-sql
opensearch-system-templates
query-insights
repository-s3


**Host/Environment (please complete the following information):**
 - Version 2.19.0

@andrross here it is, please take a look
```
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: [2025-05-06T13:32:04,994][WARN ][r.suppressed             ] [es-logs-1.srvroom.domain.net] path: /_snapshot/s3_arch_repository/logstash-2025.02.01/_restore, params: {repository=s3_arch_repository, snapshot=logstash-2025.02.01}
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: org.opensearch.transport.RemoteTransportException: [es-logs-3.srvroom.domain.net][10.250.0.4:9300][cluster:admin/snapshot/restore]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "org.opensearch.snapshots.SnapshotInfo.isRemoteStoreIndexShallowCopyEnabled()" is null
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.repositories.blobstore.BlobStoreRepository.loadShardSnapshot(BlobStoreRepository.java:4714) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.repositories.blobstore.BlobStoreRepository.getShardSnapshotStatus(BlobStoreRepository.java:4497) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.snapshots.RestoreService$1.execute(RestoreService.java:579) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.cluster.ClusterStateUpdateTask.execute(ClusterStateUpdateTask.java:67) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.cluster.service.MasterService.executeTasks(MasterService.java:917) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:469) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.cluster.service.MasterService.runTasks(MasterService.java:329) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.cluster.service.MasterService$Batcher.run(MasterService.java:229) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:206) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:264) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:955) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedOpenSearchThreadPoolExecutor.java:283) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedOpenSearchThreadPoolExecutor.java:246) ~[opensearch-2.19.0.jar:2.19.0]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
May 06 13:32:04 srvroom-es-logs-1 docker[14843]: at java.lang.Thread.run(Thread.java:1583) [?:?]

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -37,6 +37,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Fix object field exists query ([#17843](https://github.com/opensearch-project/OpenSearch/pull/17843))
 - Use Bad Request status for InputCoercionEcception ([#18161](https://github.com/opensearch-project/OpenSearch/pull/18161))
 - Null check field names in QueryStringQueryBuilder ([#18194](https://github.com/opensearch-project/OpenSearch/pull/18194))
+- Avoid NPE if on SnapshotInfo if 'shallow' boolean not present ([#18187](https://github.com/opensearch-project/OpenSearch/issues/18187))
 
 ### Security
 

--- a/server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java
+++ b/server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java
@@ -4568,7 +4568,7 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
             SnapshotId snapshotId = snapshotInfo.snapshotId();
             if (snapshotInfo.getPinnedTimestamp() != 0) {
                 return () -> IndexShardSnapshotStatus.newDone(0L, 0L, 0, 0, 0, 0, "1");
-            } else if (snapshotInfo.isRemoteStoreIndexShallowCopyEnabled()) {
+            } else if (Boolean.TRUE.equals(snapshotInfo.isRemoteStoreIndexShallowCopyEnabled())) {
                 if (shardContainer.blobExists(REMOTE_STORE_SHARD_SHALLOW_COPY_SNAPSHOT_FORMAT.blobName(snapshotId.getUUID()))) {
                     return REMOTE_STORE_SHARD_SHALLOW_COPY_SNAPSHOT_FORMAT.read(
                         shardContainer,

